lib/JsonpMainTemplatePlugin.js

```diff
@@ -114,15 +121,17 @@ JsonpMainTemplatePlugin.prototype.apply = function(mainTemplate) {
 		]);
 	});
 	mainTemplate.plugin("bootstrap", function(source, chunk, hash) {
-		if(chunk.chunks.length > 0) {
+		if(needChunkLoadingCode(chunk)) {
 			var jsonpFunction = this.outputOptions.jsonpFunction;
 			return this.asString([
 				source,
 				"",
 				"// install a JSONP callback for chunk loading",
-				"var parentJsonpFunction = window[" + JSON.stringify(jsonpFunction) + "];",
-				"window[" + JSON.stringify(jsonpFunction) + "] = function webpackJsonpCallback(chunkIds, moreModules, executeModules) {",
+				"var jsonpArray = window[" + JSON.stringify(jsonpFunction) + "] = window[" + JSON.stringify(jsonpFunction) + "] || [];",
+				"jsonpArray = jsonpArray.slice();",
```

Comment:
This makes a copy of the global `jsonpArray`, so any async additions can't be seen.